### PR TITLE
update for StrOrPromise in typeddjango/django-stubs

### PR DESCRIPTION
**Pull request by terencehonles**
# I have made things!

`StrOrPromise` updates after https://github.com/typeddjango/django-stubs/pull/689

### CODE REVIEW

**Review by intgr — CHANGES_REQUESTED** [review]
Thanks! I rebased this to fix the CI issues on master.

But a few `.yml` test cases need to be updated in this branch as well, since mypy messages changed.

File: rest_framework-stubs/fields.pyi [hunk 1]
```diff
@@ -16,6 +16,8 @@ from rest_framework.serializers import BaseSerializer
 from rest_framework.validators import Validator
 from typing_extensions import Final, TypeAlias
 
+from django_stubs_ext import StrOrPromise
```
Comment (intgr) [inline rest_framework-stubs/fields.pyi:19]: Ah also: please add explicit `django-stubs-ext` dependency to `setup.py`.

**Comment by piotrszyma** [comment]
> @terencehonles Are you up to it to fix the CI here?

It seems like author has abandoned the PR and project I maintain needs DRF stubs bump - I've fixed tests and created a new PR here https://github.com/typeddjango/djangorestframework-stubs/pull/331

**Review by intgr — APPROVED** [review]
Great! I've force pushed @piotrszyma 's test fixes from #331. I'll add a `Co-authored-by:` tag though.

**Comment by terencehonles** [comment]
Thanks for taking care of this, I have been rather busy and didn't notice the notifications.

**Pull request merged**